### ISSUE TITLE
Option to not have Pegasus hide when game is launched

### ISSUE THREAD

**Issue by PlayingKarrde** [mmatyas/pegasus-frontend#648]
If there were an option to leave Pegasus open (ideally at a theme-able state) but frozen so it doesn't use up resources I feel like this would go a long way to aid that all-in-one feel. Right now, the desktop is shown between launching games and when the emulator/game is loaded and it breaks immersion.

This of course would only be applicable to desktop builds though.

**Comment by fastpop72** [comment]
Nice feature to add! I use debian and i see the Gnome Terminal between launch game & emulator... Would be great a black screen too...

**Comment by LondoJI** [comment]
I think this is one of those reasons why I still can't totally move from HyperSpin to Pegasus. The experience to feel you're inside a Multigame System (something like Emuelec/Batocera, or at least as Lauchbox is able to) is important for me.

Would be great if we could get a way to add a transition animation, or an image when we run a game from Pegasus. Even just a black screen could be beautiful. This way Pegasus could sleep/hibernate as usual.

**Comment by notNSANE** [comment]
idea: add Pegasus to the systray and minimize it there some seconds after launching a game?

**Comment by djbobw** [comment]
I think an option to keep the frontend open is needed on Steam Deck. Steam launches Pegasus, Pegasus launches the selected game, but then closes itself, making Steam think it is finished. It therefore returns to the Steam screen, but I can hear the game I selected running in the background.

**Comment by mmatyas** [comment]
I think the Steam Deck issue might be the same as https://github.com/mmatyas/pegasus-frontend/issues/442.

**Comment by BoxWithoutATop** [comment]
Is this on the radar? This is a feature I want so I might just make a pull request. It doesn't really make much sense to me for the minimize behavior to be baked into the launch method. 

Maybe instead of baking it in we have a separate minimize method that can be called whenever, then if people want the minimize function they can just call background() or whatever in the same block that starts the application? Or have an optional bool within launch to control the minimization? Like a game.launch(false) option for those who want to disable that functionality in their themes.

**Comment by mmatyas** [comment]
Pegasus minimizes itself when the operating system reports that the launched program has started. This is required, because on some platforms strictly one program can draw to the screen at a time. If it takes some time for the launched program to show something (understandable), that is not something Pegasus can know beforehand.

The ideal solution would be probably to stay around for a few seconds _if the platform supports it_, which should be possible to add.

**Comment by BoxWithoutATop** [comment]
I think trying to detect which platform supports it and which doesn't would be unnecessarily difficult, and still limits the flexibility of the end user. Take me for example. I am running this on an HTPC in my living room. That will be more than strong enough to run Pegasus in the background, but some of the apps will still take time to boot. If the 'hang around' timer is hard-coded, this will inevitably lead to Pegasus closing before the app fully boots for at least a few items in the library, and prevents me from adding functionality that I deem valuable, like a custom 'app launching' page.

Here is what I want to do:

- Add a setting to Pegasus that toggles the post launch behavior. Call it "Pegasus Closes on Game Launch" or something.
- Expose a new signal: `postLaunch`.
- Expose another new method: `minimize()`. When called, this runs all the same default minimization behavior we currently see when `game.launch()` is run.

When `game.launch()` runs, Pegasus will signal on `postLaunch` if the user has set "Pegasus Closes on Game Launch" to false, and will run `minimize()` if the user has set it to true. By listening for `postLaunch` developers can define special behaviors, like custom UI elements, timers that run `minimize()` after a certain amount of time, or whatever else they may deem valuable.

Then the user can have either, depending on what they want. What do you think?

**Comment by mmatyas** [comment]
Whether Pegasus can run in the background doesn't depend on strength but the video driver and platform requirements. If it only allows one application, then Pegasus _must_ hide. This is known in build time though, so it's shouldn't be hard to have platform specific behavior.

I agree there could be a setting, like the time to wait before minimizing. I disagree with the theme changes though. First, it's not the theme's job or responsibility to decide when it's safe to destroy the program's window. Second, program launching is near instantaneous; the issue is when does the program appear, and Pegasus has no control over that. There's not much a `postLaunch` can do, as it'd be just a few frames ahead of launching. But we don't need it either: there are themes already with fade out animations, they just put a delay between the user interaction and the call to `game.launch()`.

This issue isn't that complex, we just need a configurable delay between receiving the program-has-launched signal, and hiding Pegasus' window, depending on the platform. I don't think any other change is necessary.

**Comment by BoxWithoutATop** [comment]
You are right and I fully agree. I do think that the option to disable the closing should still be there for those who could use it, maybe by setting the timeout to -1. Regardless, I am going to get started on this. I will submit a PR when I am done.
